gobblin-utility/src/main/java/gobblin/util/PasswordUtils.java

```diff
@@ -0,0 +1,177 @@
+   */
+  public static String encryptPassword(Properties props, String plain) {
+    String key = getKey(PropertyUtils.getStateFromProperties(props));
+    return encryptPassword(key, plain);
+  }
+
+  private static String encryptPassword(String key, String plain) {
+    StrongTextEncryptor ec = new StrongTextEncryptor();
+    ec.setPassword(key);
+    return "ENC(" + ec.encrypt(plain) + ")";
+  }
+
+  /**
+   * Decrypt a password.
+   *
+   * @param fs A FileSystem instance.
+   * @param keyLoc Location of the encryption key.
+   * @param encrypted The encrypted password.
+   * @return The decrypted password.
+   */
+  public static String decryptPassword(FileSystem fs, Path keyLoc, String encrypted) {
+    String key = getKey(fs, keyLoc);
+    return decryptPassword(key, encrypted);
+  }
+
+  /**
+   * Decrypt a password.
+   *
+   * @param state State should contain property "encrypt.key.loc", and optionally, "encrypt.key.fs.uri".
+   * If the latter is not present, "fs.uri" is used. If neither is present, "file:///" is used.
+   * @param encrypted The encrypted password.
+   * @return The decrypted password.
+   */
+  public static String decryptPassword(State state, String encrypted) {
+    String key = getKey(state);
+    return decryptPassword(key, encrypted);
+  }
+
+  /**
+   * Decrypt a password.
+   *
+   * @param props Props should contain property "encrypt.key.loc", and optionally, "encrypt.key.fs.uri".
+   * If the latter is not present, "fs.uri" is used. If neither is present, "file:///" is used.
+   * @param encrypted The encrypted password.
+   * @return The decrypted password.
+   */
+  public static String decryptPassword(Properties props, String encrypted) {
+    String key = getKey(PropertyUtils.getStateFromProperties(props));
+    return decryptPassword(key, encrypted);
+  }
+
+  private static String decryptPassword(String key, String encrypted) {
+    Matcher matcher = PASSWORD_PATTERN.matcher(encrypted);
+    Preconditions.checkArgument(matcher.matches(),
+        String.format("Encrypted password %s doesn't match pattern %s", encrypted, PASSWORD_PATTERN));
+    StrongTextEncryptor ec = new StrongTextEncryptor();
+    ec.setPassword(key);
+    try {
+      return ec.decrypt(matcher.group(1));
+    } catch (Exception e) {
+      throw new RuntimeException("Cannot decrypt password " + encrypted);
+    }
+  }
+
+  private static String getKey(State state) {
+    Preconditions.checkArgument(state.contains(ConfigurationKeys.ENCRYPT_KEY_LOC));
+    String fsUri;
+    if (state.contains(ConfigurationKeys.ENCRYPT_KEY_FS_URI)) {
+      fsUri = state.getProp(ConfigurationKeys.ENCRYPT_KEY_FS_URI);
+    } else {
+      fsUri = state.getProp(ConfigurationKeys.FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI);
+    }
+    try {
+      return getKey(FileSystem.get(new URI(fsUri), new Configuration()),
+          new Path(state.getProp(ConfigurationKeys.ENCRYPT_KEY_LOC)));
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
```

Comment:
Please create a new RuntimeException with a message that explains that the master password cannot be read and "e" is the cause. This will make logging these kind of errors a lot easier to understand.